code/obj/item/mousetrap.dm

```diff
@@ -483,3 +491,104 @@
 		if (src.mousetrap.buttbomb && src.armed)
 			playsound(src, 'sound/voice/farts/poo2.ogg', 30, 0, 0, 1.8)
 		..()
+
+// A gimmick trap for unsuspecting people
+/obj/item/mousepunch
```

Comment:
this probably go with being a child of `/obj/item/device`, the less direct childs of `/obj/item`, the better, usually.